File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.py
class Solution:
    def average(self, salary: List[int]) -> float:
        maxValue = salary[0]
        minValue = salary[0]
        
        for i in range(len(salary)):
            maxValue=max(maxValue,salary[i])
            minValue=min(minValue,salary[i])
        
        total = 0
        for i in salary:
            if(i!=minValue and i!=maxValue):
                total += i
        return total/(len(salary)-2)
            
        
        

File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.java
class Solution {
    public double average(int[] salary) {
        int minSalary = Integer.MAX_VALUE;
        int maxSalary = Integer.MIN_VALUE;
        for(int i=0; i<salary.length; i++){
            minSalary = Math.min(minSalary,salary[i]);
            maxSalary = Math.max(maxSalary,salary[i]);
        }
        double sum=0;
        for(int j=0; j<salary.length;j++){
            if(salary[j]!=maxSalary && salary[j]!=minSalary){
                sum+=salary[j];
            }
        }
        return sum/(salary.length-2);
        
    }
}